jax/numpy/lax_numpy.py

```diff
@@ -3150,6 +3151,18 @@ def sort(a, axis=-1, kind='quicksort', order=None):
   else:
     return lax.sort(a, dimension=_canonicalize_axis(axis, ndim(a)))
 
+@_wraps(np.sort_complex)
+def sort_complex(a):
+    a.sort()
+    if not issubclass(a.dtype.type, _nx.complexfloating):
```

Comment:
For GPU and TPU, Jax requires some special-casing of complex types. You can duplicate the logic found here: https://github.com/google/jax/blob/b187663a87cc2a46317e208b1810b4d65cb19916/jax/lax/lax_fft.py#L38-L44